### PR TITLE
Rework packaging

### PR DESCRIPTION
**Pull request by irismessage**
Mainly just wanted to bring pyperclip up to date but I got carried away and made a bunch of minor improvements across the board

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog for `passwdgen`
 
+## `v0.3.2` - 22 April 2023
```
Comment (thanethomson) [inline CHANGELOG.md]: Given that this change involves a toolchain upgrade, is it not worth releasing it as v0.4.0?

File: LICENSE [hunk 2]
```diff
@@ -1,23 +1,21 @@
-# The MIT License (MIT)
+The MIT License (MIT)
 
-Copyright (c) 2016 Thane Thomson
+Copyright (c) 2023
```
Comment (thanethomson) [inline LICENSE]:
```suggestion
Copyright (c) 2016- Thane Thomson and contributors
```

The start date from which copyright is valid should be indicated - not the current date. Also, afaik, most countries recognize that copyright of each individual contribution belongs to the specific contributor.

File: passwdgen/__init__.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
 # -*- coding: utf-8 -*-
 
+__version__ = "0.3.2"
```
Comment (thanethomson) [inline passwdgen/__init__.py]:
```suggestion
__version__ = "0.4.0"
```

**Review by thanethomson — COMMENTED** [review]
Thanks so much for this @joelsgp! Apologies for the delay in review, it's been a super busy week. Just some minor comments and a question around the versioning, but otherwise this looks great!

File: LICENSE (hunk 2 above)
Comment (irismessage) [reply LICENSE]: `flit init` did that automatically, thanks for noticing it :+1:

File: CHANGELOG.md (hunk 1 above)
Comment (irismessage) [reply CHANGELOG.md]: 6062673

**Review by thanethomson — APPROVED** [review]

**Pull request merged**